packages/babel-generator/src/buffer.js

```diff
@@ -101,7 +101,7 @@ export default class Buffer {
 
   rightBrace() {
     this.newline(true);
-    if (this.format.compact && !this._lastPrintedIsEmptyStatement) {
+    if (this.format.minified && !this._lastPrintedIsEmptyStatement) {
```

Comment:
Yeah. I think leaving out semicolons is dangerous, especially with our
implementation which is a bit hacky.

On Wednesday, December 9, 2015, James Kyle <notifications@github.com> wrote:

> In packages/babel-generator/src/buffer.js
> <https://github.com/babel/babel/pull/3152#discussion_r47190011>:
>
> > @@ -101,7 +101,7 @@ export default class Buffer {
> >
> >    rightBrace() {
> >      this.newline(true);
> > -    if (this.format.compact && !this._lastPrintedIsEmptyStatement) {
> > +    if (this.format.minified && !this._lastPrintedIsEmptyStatement) {
>
> How are we differentiating compact and minified? I was hoping the more
> minified mode would just be enabling the more dangerous stuff.
>
> The compact description says:
>
> do not include superfluous whitespace characters and line terminators
> [true|false|auto]
>
> which seems to include this
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/babel/babel/pull/3152/files#r47190011>.
>